hs_core/management/commands/update_res_files_system_metadata.py

```diff
@@ -0,0 +1,45 @@
+from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist
+from django.core.management.base import BaseCommand, CommandError
+
+from hs_core.hydroshare import get_resource_by_shortkey
+from hs_core.models import ResourceFile
+
+
+class Command(BaseCommand):
+    help = "Updates modified_time, checksum, size for all files of a resources in DB for a given resource"
+
+    def add_arguments(self, parser):
+        parser.add_argument('--resource_id', type=str, required=True,
```

Comment:
this management command looks good.
I understand that updating all existing files on production would take too long (~3days we estimate).
That means we would be proceeding with most files currently on the system having None for their file_system_metadata (checksum and modified time).

I see that these properties are calculated on the fly, if they aren't already stored on the ResourceFile
https://github.com/hydroshare/hydroshare/blob/5182-store-file-checksum-modified-time-in-db/hs_core/models.py#L3105-L3173

So just confirming that our plan is to calculate them on the fly for now and only use this management command when someone complains about a specific resource?

Reply:
That is correct.

Reply:
Nice sounds good thanks for confirming! We could create a task to run this management command for a subset of resources every night, gradually creating all of the missing checksums over the course of a few weeks... just a thought.

Reply:
If you think that is a good idea but don't have time to implement, LMK and I would be happy to add it

Reply:
@devincowan I think it would be good to run a nightly job to do this update of existing resources. It would be great if you can add it.

Reply:
okay I will add a nightly job this evening

Reply:
@pkdash added this in https://github.com/hydroshare/hydroshare/pull/5184/commits/365b8e34261424aa1b67e8e998b3fa490e93d5a6